src/main/java/Duke.java

```diff
@@ -1,10 +1,93 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
+        int count = 0;
+        String input;
+        Task list[] = new Task[100];
+
+        printWelcomeMessage();
+
+        // Echo loops until "bye" is inputted
+        while(true) {
+            Scanner echo = new Scanner(System.in); // Scan and echo user input
+            input = echo.nextLine();
+
+            // Exit program if "bye" is inputted
+            if (input.equals("bye")) {
+                System.out.println("Goodbye. See you soon!");
+                break;
+            }
+
+            else if (input.equals("list")){
+                printList(list);
+            }
+
+            else if (input.contains("done")){
+                markAsDone(input, list);
+            }
+
+            // Add input as new task
+            else {
+                count = addTask(count, input, list);
+            }
+        }
+    }
+
+    private static int addTask(int count, String input, Task[] list) {
+        if (input.contains("deadline")){
+            int slashLocation = input.indexOf("/");
+            list[count] = new Deadline(input.substring(input.indexOf("n") + 3,slashLocation), input.substring(slashLocation+1));
```

Comment:
Perhaps can add a few more variables to store the processed message before this line of code. This can decrease the complexity of the expressions and improve the readability.  I noticed the same issue in several other parts too.